server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -0,0 +1,242 @@
+            builder.field("status", "green");
+            builder.field("cluster_name", clusterName.value());
+            builder.array("impacts");
+            builder.startObject("components");
+            for (Component component : components) {
+                builder.startObject(component.getName());
+                builder.field("status", component.getStatus());
+                builder.startArray("indicators");
+                List<Indicator> indicators = component.getIndicators();
+                for (Indicator indicator : indicators) {
+                    builder.startObject();
+                    builder.field("name", indicator.getName());
+                    builder.field("status", indicator.getStatus());
+                    builder.field("explain", indicator.getExplain());
+                    builder.startObject("meta");
+                    indicator.toXContent(builder, params);
+                    builder.endObject();
+                    // TODO: Add detail / documentation
+                    builder.endObject();
+                }
+                builder.endArray();
+                builder.endObject();
+            }
+            builder.endObject();
+            return builder.endObject();
+        }
+    }
+
+    public static class Request extends ActionRequest {
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+    }
+
+    public static class TransportAction extends org.elasticsearch.action.support.TransportAction<Request, Response> {
+
+        private final ClusterService clusterService;
+
+        @Inject
+        public TransportAction(
+            final ActionFilters actionFilters,
+            final TransportService transportService,
+            final ClusterService clusterService
+        ) {
+            super(NAME, actionFilters, transportService.getTaskManager());
+            this.clusterService = clusterService;
+        }
+
+        @Override
+        protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+            final ClusterState clusterState = clusterService.state();
+            final Controller component = new Controller(clusterService.localNode(), clusterState);
+            listener.onResponse(new Response(clusterService.getClusterName(), Arrays.asList(component, new Backups())));
+        }
+    }
+
+    private interface Component {
+
+        String getName();
+
+        ClusterHealthStatus getStatus();
+
+        List<Indicator> getIndicators();
+
+    }
+
+    private interface Indicator extends ToXContentFragment {
+
+        String getName();
+
+        ClusterHealthStatus getStatus();
+
+        String getExplain();
+
+    }
+
+    private record NodeDoesNotHaveMaster(DiscoveryNode node, DiscoveryNode masterNode) implements Indicator {
+
+        public static final String GREEN_EXPLAIN = "health coordinating instance has a master node";
```

Comment:
Will there be an easy way to review all of the explain texts? Is there a good way to ensure that all Indicators follow similar patterns in tracking Green/Red explain texts so that these messages can be easily understood/identified? As it is now, without developer consistency on following patterns for constants, I would need to interpret each `getExplain` to find all the potential Strings that are referenced in that method, which could get unwieldy to interpret over time.

Reply:
My current understanding is that each indicator will be describing only single possible issue. This means explain would be purely human readable message and it will be possible to use indicator name (much easier to find all by looking on available implementations) to judge the failure area.

Reply:
@elastichelix it sounds like you need to understand these texts? If so, we should probably discuss for what purpose you need this.

Besides the text, we will provide an `id` of the indicator as well as a bit of data. Those are probably better for programmatic use.

But the intention is that you consume the `explain` text as the high-level one line summary and then drill into specific health issues (of which there can be many for one indicator).